packages/vite/src/utils/warmup.ts

```diff
@@ -50,7 +54,10 @@ export async function warmupViteServer (
       const m = await server.moduleGraph.getModuleByUrl(url, isServer)
       // a module that is already compiled (and can't be warmed up anyway)
       if (m?.transformResult?.code || m?.ssrTransformResult?.code) {
```

Comment:
Maybe we could check `isServer ? m?.ssrTransformResult?.code : m?.transformResult?.code`?